Incorrect OpenGL version detected on desktop

**Describe the bug**

I was trying to figure out why I couldn't use feature level > 1 and I think I found some hints.
queryOpenGLVersion returns 4.1 if WGL_CONTEXT_MINOR_VERSION_ARB is set to 1 in PlatformWGL.cpp (wglCreateContextAttribs call), and 4.6 if I set it to 6.

I got the idea to try this after I found a similar issue in another code base : https://github.com/mpv-player/mpv/commit/0e76372e86d92982e17219fda6597f92ca67e7e8

/*
     * Nvidia drivers will not provide the highest supported version
     * when 320 core is requested. Instead, it just returns 3.2. This
     * would be bad, as we actually want compute shaders that require
     * 4.2, so we have to request a sufficiently high version. We use
     * 440 to maximise driver compatibility as we don't need anything
     * from newer versions.
     */

Looking at how they initialize GL, it seems that they try higher versions first :

https://github.com/mpv-player/mpv/blob/0d44ae319dfd5209eb0733caafdcbcf67d228cf3/video/out/opengl/context_glx.c

Maybe it's an idea for a proper fix ?

**Desktop (please complete the following information):**
 - OS: Windows
 - GPU: NVIDIA GTX 1080
 - Backend: OpenGL

Hello,

I cherry picked commit 57a8e0cf01999044748116130a96e3f0c4d6d0fa

It doesn't compile : 
> PlatformWGL.cpp(125,9): error C2362: initialization of 'attribs' is skipped by 'goto error'
It looks like the issue was picked up by your CI.

Moving the declaration of attribs above, then it does initialize the engine at feature level 3.
I'm not sure why you start at 4.5 and not above though ? To future proof the init.

--- a/filament/backend/src/opengl/platforms/PlatformWGL.cpp
+++ b/filament/backend/src/opengl/platforms/PlatformWGL.cpp
@@ -99,12 +99,6 @@ Driver* PlatformWGL::createDriver(void* const sharedGLContext,
         0, 0, 0
     };
 
-    int attribs[] = {
-        WGL_CONTEXT_MAJOR_VERSION_ARB, 4,
-        WGL_CONTEXT_MINOR_VERSION_ARB, 1,
-        0
-    };
-
     HGLRC tempContext = NULL;
 
     mHWnd = CreateWindowA("STATIC", "dummy", 0, 0, 0, 1, 1, NULL, NULL, NULL, NULL);
@@ -127,7 +121,20 @@ Driver* PlatformWGL::createDriver(void* const sharedGLContext,
 
     wglCreateContextAttribs =
             (PFNWGLCREATECONTEXTATTRIBSARBPROC) wglGetProcAddress("wglCreateContextAttribsARB");
-    mContext = wglCreateContextAttribs(whdc, (HGLRC) sharedGLContext, attribs);
+
+    // try all versions down, from GL 4.5 to 4.1
+    for (int minor = 5; minor >= 1; minor--) {
+        int const attribs[] = {
+                WGL_CONTEXT_MAJOR_VERSION_ARB, 4,
+                WGL_CONTEXT_MINOR_VERSION_ARB, minor,
+                0
+        };
+        mContext = wglCreateContextAttribs(whdc, (HGLRC)sharedGLContext, attribs);
+        if (mContext) {
+            break;
+        }
+    }
+
     if (!mContext) {
         utils::slog.e << "wglCreateContextAttribs() failed, whdc=" << whdc << utils::io::endl;
         goto error;

